frontend/src/components/phones/onboarding/RelayNumberPicker.tsx

```diff
@@ -82,29 +84,76 @@ const RelayNumberIntro = (props: RelayNumberIntroProps) => {
 
 type RelayNumberSelectionProps = {
   registerRelayNumber: (phoneNumber: string) => Promise<Response>;
+  search: (search: string) => Promise<Response | undefined>;
 };
 const RelayNumberSelection = (props: RelayNumberSelectionProps) => {
   const { l10n } = useLocalization();
   const relayNumberSuggestionsData = useRelayNumberSuggestions();
   const [phoneNumber, setPhoneNumber] = useState("");
+  const [searchValue, setSearchValue] = useState("");
   const [relayNumberIndex, setRelayNumberIndex] = useState(0);
-  const relayNumberSuggestions = relayNumberSuggestionsData.data
-    ? [
+  const [isSearching, setIsSearching] = useState(false);
+  const [relayNumberSuggestions, setRelayNumberSuggestions] = useState<
+    string[]
+  >([]);
+
+  if (relayNumberSuggestionsData.data && relayNumberSuggestions.length === 0) {
+    setRelayNumberSuggestions(
+      [
         ...relayNumberSuggestionsData.data.same_area_options,
         ...relayNumberSuggestionsData.data.other_areas_options,
         ...relayNumberSuggestionsData.data.same_prefix_options,
       ].map((suggestion) => suggestion.phone_number)
-    : [];
+    );
+  }
 
-  const onChange: ChangeEventHandler<HTMLInputElement> = (event) => {
+  const onRelayNumberChange: ChangeEventHandler<HTMLInputElement> = (event) => {
     setPhoneNumber(event.target.value);
   };
 
-  const onSubmit: FormEventHandler = (event) => {
+  const onSearchChange: ChangeEventHandler<HTMLInputElement> = (event) => {
+    setSearchValue(event.target.value);
+  };
+
+  const onSubmitRelayNumber: FormEventHandler = (event) => {
     event.preventDefault();
     props.registerRelayNumber(phoneNumber);
   };
 
+  const onSubmitSearch: FormEventHandler = (event) => {
+    event.preventDefault();
+
+    // return early if search is currently taking place
+    if (isSearching) return;
+
+    // set search to true as we begin searching
+    setIsSearching(true);
+
+    // submit string for search to our API
+    props
+      .search(searchValue)
+      .then((res) => res?.json())
+      .then((data) => {
```

Comment:
Non-blocking suggestion: `data` is untyped here (because `.json()` resolves to `any`), which is a bit of a shame. Repercussions are limited, since it's only used here, but ideally, I'd like to see `search` to call `.json()` by itself and define a type representing that data, rather than returning an opaque `Response`.  That way, if we ever find we need to do validation on the response value, we can do so in one place and have TypeScript tell us if that breaks anywhere that `.search` is used, rather than having to do a search across the codebase and manually read and understand all usage sites.